### ISSUE TITLE
Suddenly crashing on parts

### ISSUE THREAD

**Issue by rlpowell** [zerowidth/camper_van#40]
This has suddenly started happening to me a lot:

 /usr/local/rvm/gems/ruby-1.9.3-p392/gems/camper_van-0.0.11/lib/camper_van/channel.rb:90:in `part': undefined method`close_connection' for #EventMachine::HttpClient:0x00000004657fa8 (NoMethodError)
        from /usr/local/rvm/gems/ruby-1.9.3-p392/gems/camper_van-0.0.11/lib/camper_van/ircd.rb:285:in `block (2 levels) in <class:IRCD>'
        from /usr/local/rvm/gems/ruby-1.9.3-p392/gems/camper_van-0.0.11/lib/camper_van/ircd.rb:284:in`each'
        from /usr/local/rvm/gems/ruby-1.9.3-p392/gems/camper_van-0.0.11/lib/camper_van/ircd.rb:284:in `block in <class:IRCD>'
        from /usr/local/rvm/gems/ruby-1.9.3-p392/gems/camper_van-0.0.11/lib/camper_van/command_definition.rb:59:in`handle'
        from /usr/local/rvm/gems/ruby-1.9.3-p392/gems/camper_van-0.0.11/lib/camper_van/ircd.rb:77:in `receive_line'
        from /usr/local/rvm/gems/ruby-1.9.3-p392/gems/camper_van-0.0.11/lib/camper_van/server.rb:126:in`receive_line'
        from /usr/local/rvm/gems/ruby-1.9.3-p392/gems/eventmachine-1.0.3/lib/em/protocols/linetext2.rb:64:in `receive_data'
        from /usr/local/rvm/gems/ruby-1.9.3-p392/gems/eventmachine-1.0.3/lib/eventmachine.rb:187:in`run_machine'
        from /usr/local/rvm/gems/ruby-1.9.3-p392/gems/eventmachine-1.0.3/lib/eventmachine.rb:187:in `run'
        from /usr/local/rvm/gems/ruby-1.9.3-p392/gems/camper_van-0.0.11/lib/camper_van/server.rb:23:in`run'
        from /usr/local/rvm/gems/ruby-1.9.3-p392/gems/camper_van-0.0.11/bin/camper_van:75:in `<top (required)>'
        from /usr/local/rvm/gems/ruby-1.9.3-p392/bin/camper_van:19:in`load'
        from /usr/local/rvm/gems/ruby-1.9.3-p392/bin/camper_van:19:in `<main>'
        from /usr/local/rvm/gems/ruby-1.9.3-p392/bin/ruby_noexec_wrapper:14:in`eval'
        from /usr/local/rvm/gems/ruby-1.9.3-p392/bin/ruby_noexec_wrapper:14:in `<main>'

**Comment by mmichie** [comment]
I'm seeing this issue as well

**Comment by pbrisbin** [comment]
Relevant lines [here](https://github.com/zerowidth/camper_van/blob/master/lib/camper_van/channel.rb#L89-91).

**Comment by pastorius** [comment]
I'm seeing this too on 0.0.12. 

@zerowidth any ideas on where to start digging in on this? The EM docs clearly include `EventMachine::HttpClient#close_connection`.

I'm happy to take a look at it, but could use a little guidance.

**Comment by lovek323** [comment]
I'm experiencing the same issue. Any assistance would be appreciated.

**Comment by rlpowell** [comment]
FWIW, I simply commented out the line in question, and everything works fine.

**Comment by lovek323** [comment]
@rlpowell -- that has solved the problem, but I can't work out why the method doesn't exist. The `HttpClient` class does inherit from the `Connection` class which responds to the `close_connection` message.

**Comment by jimeh** [comment]
I've got the same issue, seems to happen whenever a user disconnects. For now I'll just stick to 0.0.10, as this issue occurs with 0.0.11 and 0.0.12.

```
2013-08-10 09:56:27     1  INFO CamperVan : closed connection from 192.168.50.1
2013-08-10 09:56:27     1 ERROR CamperVan : http error . Trying again in 2 seconds...
/var/lib/gems/1.9.1/gems/camper_van-0.0.12/lib/camper_van/channel.rb:90:in `part': undefined method `close_connection' for #<EventMachine::HttpClient:0x00000000d31620> (NoMethodError)
        from /var/lib/gems/1.9.1/gems/camper_van-0.0.12/lib/camper_van/ircd.rb:285:in `block (2 levels) in <class:IRCD>'
        from /var/lib/gems/1.9.1/gems/camper_van-0.0.12/lib/camper_van/ircd.rb:284:in `each'
        from /var/lib/gems/1.9.1/gems/camper_van-0.0.12/lib/camper_van/ircd.rb:284:in `block in <class:IRCD>'
        from /var/lib/gems/1.9.1/gems/camper_van-0.0.12/lib/camper_van/command_definition.rb:59:in `handle'
        from /var/lib/gems/1.9.1/gems/camper_van-0.0.12/lib/camper_van/ircd.rb:77:in `receive_line'
        from /var/lib/gems/1.9.1/gems/camper_van-0.0.12/lib/camper_van/server.rb:126:in `receive_line'
        from /var/lib/gems/1.9.1/gems/eventmachine-1.0.3/lib/em/protocols/linetext2.rb:64:in `receive_data'
        from /var/lib/gems/1.9.1/gems/eventmachine-1.0.3/lib/eventmachine.rb:187:in `run_machine'
        from /var/lib/gems/1.9.1/gems/eventmachine-1.0.3/lib/eventmachine.rb:187:in `run'
        from /var/lib/gems/1.9.1/gems/camper_van-0.0.12/lib/camper_van/server.rb:23:in `run'
        from /var/lib/gems/1.9.1/gems/camper_van-0.0.12/bin/camper_van:75:in `<top (required)>'
        from /usr/local/bin/camper_van:19:in `load'
        from /usr/local/bin/camper_van:19:in `<main>'
```

**Comment by zerowidth** [comment]
Fixed in #44. Thanks @tleish!
